src/player.cpp

```diff
@@ -3162,12 +3162,8 @@ bool Player::updateSaleShopList(const Item* item)
 				return false;
 			}
 
-			for (uint32_t index = 0; index < container->size(); ++index) {
-				if (updateSaleShopList(container->getItemByIndex(index))) {
-					return true;
-				}
-			}
-			return false;
+			const auto& items = container->getItemList();
+			return std::any_of(items.begin(), items.end(), [this](const Item* item){return updateSaleShopList(item);});
```

Comment:
Just a style nitpick, this is more readable to me (also renamed item to containerItem to not shadow the item parameter if we ever would use it in the lambda in the future):
```cpp
return std::any_of(items.begin(), items.end(), [this](const Item* containerItem) {
	return updateSaleShopList(containerItem);
});
```